decam_process.py

```diff
@@ -0,0 +1,266 @@
+        refObjLoader.retarget(LoadIndexedReferenceObjectsTask)
+        refObjLoader.ref_dataset_name = 'pan-starrs'  # options are gaia, pan-starrs, sdss
+        refObjLoader.filterMap = {"g": "g",  # 'phot_g_mean_mag' is gaia-specific
+                                  "r": "r",  # all of 'g,r,i,z,y' are options for pan-starrs
+                                  "VR": "g"}
+    args = [repo, '--id', 'visit=' + visit, 'ccdnum=' + ccdnum,
+            '--output', processedrepo,
+            '--calib', calibrepo,
+            '-C', OBS_DECAM_DIR + '/config/processCcdCpIsr.py',
+            '--config', 'calibrate.doAstrometry=True',
+            'calibrate.doPhotoCal=True']
+    ProcessCcdTask.parseAndRun(args=args, config=config)
+    return
+
+
+def doDiffIm(processedrepo, sciencevisit, ccdnum, templatevisit, diffimrepo):
+    '''
+    Do difference imaging with a visit as a template and one or more as science
+
+    USAGE:
+    $ python hits_ingest.py diffIm
+
+    BASH EQUIVALENT:
+    $ imageDifference.py processedrepo --id visit=sciencevisit ccdnum=ccdnum
+            --templateId visit=templatevisit --output diffimrepo
+            -C diffim_config.py
+    ** to run from bash, 'diffim_config.py' must exist and contain, e.g.,
+        from lsst.ip.diffim.getTemplate import GetCalexpAsTemplateTask
+        config.getTemplate.retarget(GetCalexpAsTemplateTask)
+        config.detection.thresholdValue=5.0
+        config.doDecorrelation=True
+
+    RESULT:
+    diffimrepo/deepDiff/v+sciencevisit populated with difference images
+    and catalogs of detected sources (diaSrc, diffexp, and metadata files)
+    '''
+    if not os.path.isdir(diffimrepo):
+        os.mkdir(diffimrepo)
+    print('Running ImageDifference...')
+    config = ImageDifferenceConfig()
+    config.getTemplate.retarget(GetCalexpAsTemplateTask)
+    config.detection.thresholdValue = 5.0
+    config.doDecorrelation = True
+    args = [processedrepo, '--id', 'visit=' + sciencevisit, 'ccdnum=' + ccdnum,
+            '--templateId', 'visit=' + templatevisit, '--output', diffimrepo]
+    ImageDifferenceTask.parseAndRun(args=args, config=config)
+    return
+
+
+if __name__ == '__main__':
+    # parse command line arguments with argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument('task', help='Which task you would like to run',
+                        choices=['ingest', 'ingestCalibs', 'processCcd', 'diffIm'])
+    parser.add_argument('-f', '--files', help='Input files or directory')
+    args = parser.parse_args()
+    if (args.task == 'ingest' or args.task == 'ingestCalibs') and not args.files:
+        raise IOError('-f is required to ingest images or calibrations.')
+    if args.files is None:
+        datadir = None
+        datafiles = None
+    else:
+        datadir = args.files
+        if os.path.isdir(datadir):
+            datafiles = glob(os.path.join(datadir, '*.fz'))
+        else:
+            datafiles = glob(datadir)
+    main()
```

Comment:
OK, that makes sense. I'd like to have the "main stuff" upfront, including calling the functions that follow, for clarity. To be less implicit but still support this structure, I changed the call to main at the end to include _just_ `main()`.